### DISCUSSION TITLE
What is the recommended versions couple for Node.js & MongoDB for production ?

### DISCUSSION THREAD

**Discussion by ThVuillaume** [apostrophecms/apostrophe#3639] [Q&A] [answered]
## Question or comment

Hello everyone !

I have a question related to hosting Apostrophe 3 in production.
On the [setting-up documentation](https://v3.docs.apostrophecms.org/guide/setting-up.html#system-requirements) it is said that minimum versions are node 12.x or better, and mongoDB 3.6 or better, but [here in the cookbook](https://v3.docs.apostrophecms.org/cookbook/ubuntu-hosting.html#setting-up-the-ubuntu-vps) Node.js 16.x is mentioned...

But do you guys have the recommended versions for production ? Like for example you know that it is bullet-proof tested on `Node 14 + MongoDB 4` ?

Because currently i'm using Node 16LTS + Mongodb 5.0.5 without any apearing issues (well, for now) but I would like to avoid encountering troubles in production. 

Could you please point out what is the recomendations for the Node.js+mongoDB version couple ?

**Comment by boutell** [comment, marked as answer]
Good question. Node 16 and Mongo 5.x are supported and recommended. However all current LTS releases of Node (so, back to Node 12) are supported, and all supported versions of MongoDB (so, 4.0 or better, at the moment).

We should update our guidance to clarify what is best (Node 16 and Mongo 5.x), what is too bleeding edge (Node 17 and MongoDB are not friends yet), and what is tolerable but not a great choice for new work. I'll submit a doc PR.
